packages/mobile/src/send/utils.ts

```diff
@@ -189,34 +184,26 @@ export function* handleSendPaymentData(
 
   yield put(storeLatestInRecents(recipient))
 
-  if (data.currencyCode) {
-    yield put(selectPreferredCurrency(data.currencyCode as LocalCurrencyCode))
-    const action: FetchCurrentRateSuccessAction | FetchCurrentRateFailureAction = yield take([
-      Actions.FETCH_CURRENT_RATE_SUCCESS,
-      Actions.FETCH_CURRENT_RATE_FAILURE,
-    ])
-    if (action.type === Actions.FETCH_CURRENT_RATE_FAILURE) {
-      yield put(showError(ErrorMessages.EXCHANGE_RATE_FAILED))
-      Logger.warn(TAG, '@handleSendPaymentData failed to fetch current rate')
-      return
-    }
-  }
-
-  if (data.amount) {
-    // TODO: integrate with SendConfirmation component
-    const exchangeRate = yield select(getLocalCurrencyExchangeRate)
-    const amount = convertLocalAmountToDollars(data.amount, exchangeRate)
-    if (!amount) {
+  if (data.amount && data.currencyCode) {
+    const currency = data.currencyCode as LocalCurrencyCode
+    const exchangeRate: string = yield call(fetchExchangeRate, currency)
+    const dollarAmount = convertLocalAmountToDollars(data.amount, exchangeRate)
+    if (!dollarAmount) {
       Logger.warn(TAG, '@handleSendPaymentData null amount')
       return
     }
     const transactionData: TransactionDataInput = {
       recipient,
-      amount,
+      amount: dollarAmount,
       reason: data.comment,
       type: TokenTransactionType.PayPrefill,
     }
-    navigate(Screens.SendConfirmation, { transactionData, isFromScan: true })
+
+    navigate(Screens.SendConfirmation, {
+      transactionData,
+      isFromScan: true,
+      currencyInfo: { localCurrencyCode: currency, localExchangeRate: exchangeRate },
```

Comment:
From the conversation we had with @corey and @nityas, my understanding was that we would go with the solution where "Everyone sees things in their own currency—Jean requests  €100, Nitya sees a request for $110. No changes to drilldown."
In that case the `currencyInfo` override here isn't needed, right?

Reply:
oh hm, from what I understood, that's usually the case if the transaction just specifies an amount in cusd, but if the amount is given in a specific currency (ex. when scanning a QR code for a specific transaction), then we default to that currency.

Reply:
Can you check with @nityas / @coreycelo to confirm what we want?

Reply:
Seems to be what we want :) ![Screen Shot 2020-10-21 at 1 18 00 PM](https://user-images.githubusercontent.com/15326838/96754950-05ec4f00-13a0-11eb-8f3c-648fbd914558.png)

Reply:
Sounds good! let's merge this!